gcloud/bigquery/job.py

```diff
@@ -23,6 +23,254 @@
+            millis = statistics.get('creationTime')
+            if millis is not None:
+                return _datetime_from_microseconds(millis * 1000.0)
+
+    @property
+    def started(self):
+        """Datetime at which the job was started.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        :returns: the start time (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            millis = statistics.get('startTime')
+            if millis is not None:
+                return _datetime_from_microseconds(millis * 1000.0)
+
+    @property
+    def ended(self):
+        """Datetime at which the job finished.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        :returns: the end time (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            millis = statistics.get('endTime')
+            if millis is not None:
+                return _datetime_from_microseconds(millis * 1000.0)
+
+    @property
+    def error_result(self):
+        """Error information about the job as a whole.
+
+        :rtype: mapping, or ``NoneType``
+        :returns: the error information (None until set from the server).
+        """
+        status = self._properties.get('status')
+        if status is not None:
+            return status.get('errorResult')
+
+    @property
+    def errors(self):
+        """Information about individual errors generated by the job.
+
+        :rtype: list of mappings, or ``NoneType``
+        :returns: the error information (None until set from the server).
+        """
+        status = self._properties.get('status')
+        if status is not None:
+            return status.get('errors')
+
+    @property
+    def state(self):
+        """Status of the job.
+
+        :rtype: string, or ``NoneType``
+        :returns: the state (None until set from the server).
+        """
+        status = self._properties.get('status')
+        if status is not None:
+            return status.get('state')
+
+    def _require_client(self, client):
+        """Check client or verify over-ride.
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+
+        :rtype: :class:`gcloud.bigquery.client.Client`
+        :returns: The client passed in or the currently bound client.
+        """
+        if client is None:
+            client = self._client
+        return client
+
+    def _scrub_local_properties(self, cleaned):
+        """Helper:  handle subclass properties in cleaned."""
+        pass
+
+    def _set_properties(self, api_response):
+        """Update properties from resource in body of ``api_response``
+
+        :type api_response: httplib2.Response
```

Comment:
Now that we have `httplib` intersphinx, you can say 

```
:class:`httplib2.Response`
```

and it will link out to their docs. Yay!

However this is a protected method on a protected class, so it won't get rendered in the docs. But still worth knowing I think.